### PR TITLE
[CPU][Inductor] Use VecMask::from for scalar masks in codegen

### PR DESCRIPTION
**Pull request by fadara01**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.12.0) (oldest at bottom):
* __->__ #178148

Fixes: #178136, https://github.com/vllm-project/vllm/issues/37325

Signed-off-by: Fadi Arafeh <fadi.arafeh@arm.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @mlazos

### CODE REVIEW

**Review by malfet — CHANGES_REQUESTED** [review]
Agree with Claude on `explicit` decorator, or may be even fix codegen rather than veclib, so that change can be cherry-picked into 2.11.1, as this one needs to wait until 2.12

Also, please add regression test

File: aten/src/ATen/cpu/vec/vec_mask.h [hunk 1]
```diff
@@ -127,6 +127,7 @@ class VecMask {
  public:
   VecMask() : mask_(static_cast<T>(0)) {}
   VecMask(const VectorizedN<T, N>& mask) : mask_(mask) {}
+  VecMask(const T mask) : mask_(VectorizedN<T, N>(VecMask<T, N>::from(mask))) {}
```
Comment (Skylion007) [inline aten/src/ATen/cpu/vec/vec_mask.h]: Shouldn't this be an explicit ctor for safety? Implicit conversions from scalar seem like they could be dangerous

Comment (fadara01) [reply aten/src/ATen/cpu/vec/vec_mask.h]: yup, I'll fix this in codegen instead of introducing a new constructor to have it cherry-picked into 2.11.1,

File: test/inductor/test_cpu_repro.py [hunk 2]
```diff
@@ -5077,6 +5077,28 @@ def fn(x):
...
+            y = x[0].clone()
+            y[..., 1::3] = x[1, ..., 1::3]
+            y[..., 2::3] = x[2, ..., 2::3]
+            return y
+
+        positions = torch.tensor([[0, 0], [0, 0], [0, 0]], dtype=torch.int64)
+        cache = torch.arange(3, dtype=torch.float32).reshape(1, 3)
+
+        torch._dynamo.reset()
+        metrics.reset()
```
Comment (malfet) [inline test/inductor/test_cpu_repro.py]: Isn't that done by the `setUp` function of all inductor scripts?

File: test/inductor/test_cpu_repro.py [hunk 3]
```diff
@@ -5077,6 +5077,28 @@ def fn(x):
...
+        with torch.no_grad(), config.patch({"cpp.enable_loop_tail_vec": True}):
+            expected = fn(positions, cache)
+            compiled_fn = torch.compile(fn, backend="inductor", fullgraph=True)
```
Comment (malfet) [inline test/inductor/test_cpu_repro.py]: Isn't that a default? Also, isn't `fullgraph` is also a mandate for all tests?
```suggestion
            compiled_fn = torch.compile(fn, fullgraph=True)
```

File: test/inductor/test_cpu_repro.py [hunk 4]
```diff
@@ -5077,6 +5077,28 @@ def fn(x):
...
+            y[..., 1::3] = x[1, ..., 1::3]
+            y[..., 2::3] = x[2, ..., 2::3]
+            return y
+
+        positions = torch.tensor([[0, 0], [0, 0], [0, 0]], dtype=torch.int64)
+        cache = torch.arange(3, dtype=torch.float32).reshape(1, 3)
+
+        torch._dynamo.reset()
+        metrics.reset()
+        with torch.no_grad(), config.patch({"cpp.enable_loop_tail_vec": True}):
```
Comment (malfet) [inline test/inductor/test_cpu_repro.py]: Nit (nograd is not important here, is it?)
```suggestion
        with config.patch({"cpp.enable_loop_tail_vec": True}):
```

**Review by malfet — APPROVED** [review]

File: test/inductor/test_cpu_repro.py (hunk 2 above)
Comment (fadara01) [reply test/inductor/test_cpu_repro.py]: true, I shouldn't have just blindly followed pattern in the other tests

**Review by Skylion007 — APPROVED** [review]
